pandas/_libs/lib.pyx

```diff
@@ -2357,3 +2354,200 @@ def fast_multiget(dict mapping, ndarray keys, default=np.nan):
             output[i] = default
 
     return maybe_convert_objects(output)
+
+
+@cython.wraparound(False)
+@cython.boundscheck(False)
+cdef inline void convert_and_set_item(object item, Py_ssize_t index,
+                                      object[:] result,
+                                      bint keep_trivial_numbers):
+    """
+    Convert `item` to str and set into result[index].
+
+    Parameters
+    ----------
+    item : object
+    index : Py_ssize_t
+    result : memoryview of 1-d ndarray
+    keep_trivial_numbers : bool
+        if True, then conversion (to string from integer/float zero)
+        is not performed
+    """
+    cdef:
+        bint do_convert = 1
+        float64_t float_item
+
+    if keep_trivial_numbers:
+        if isinstance(item, int):
+            if <int>item == 0:
+                do_convert = 0
+        elif isinstance(item, float):
+            float_item = item
+            if float_item == 0.0 or float_item != float_item:
+                do_convert = 0
+
+    if do_convert and not isinstance(item, str):
+        item = PyObject_Str(item)
+
+    result[index] = item
+
+
+@cython.wraparound(False)
+@cython.boundscheck(False)
+cdef inline void put_object_as_unicode(list lst, Py_ssize_t idx,
+                                       object item):
+    if not isinstance(item, str):
+        item = PyObject_Str(item)
```

Comment:
same, this should just do conersion, not setting

Reply:
Removed this function.